zcash_client_sqlite/src/wallet.rs

```diff
@@ -150,28 +152,36 @@ fn parse_account_source(
     spending_key_available: bool,
     key_source: Option<String>,
 ) -> Result<AccountSource, SqliteClientError> {
-    match (account_kind, hd_seed_fingerprint, hd_account_index) {
-        (0, Some(seed_fp), Some(account_index)) => Ok(AccountSource::Derived {
-            seed_fingerprint: SeedFingerprint::from_bytes(seed_fp),
-            account_index: zip32::AccountId::try_from(account_index).map_err(|_| {
-                SqliteClientError::CorruptedData(
-                    "ZIP-32 account ID from wallet DB is out of range.".to_string(),
-                )
-            })?,
+    let derivation = hd_seed_fingerprint
+        .zip(hd_account_index)
```

Comment:
I believe here we are now relying exclusively on the table constraint to ensure that these are both either `Some` or `None`, rather than catching a mixed-state corruption below.

Reply:
Gah, I had a nice long comment here about how this now isn't as thoroughly checking for corrupt data, then a few lines below gave me the impression that we were instead relying on the table constraint so I deleted my comment, but now I see that the table constraint has been modified to remove that check. So yeah, now if one or other of these table columns gets corrupted and nulled, we will ignore whichever column is still non-null instead of detecting the corruption.

Reply:
Yeah, I thought about adding a check to ensure that the nullability of the two columns was equal, but TBH I think that the parsing catching it is adequate.

Reply:
The parsing doesn't catch it though. `Option::zip` is an AND; it maps mixed-optionals to `None`. The previous explicit matching on the two columns did catch it because we only "parsed" `(Some(_), Some(_))` or `(None, None`).

Reply:
It doesn't catch explicit corruption, but what I mean is that it provides the desired interface. I can change it to ensure that they match though.